src/vs/workbench/api/browser/mainThreadAuthentication.ts

```diff
@@ -536,4 +544,40 @@ export class MainThreadAuthentication extends Disposable implements MainThreadAu
 		}
 		return false;
 	}
+
+	async $promptForClientDetails(authorizationServerUrl: string): Promise<{ clientId: string; clientSecret?: string } | undefined> {
```

Comment:
I think you should show a modal dialog first saying "Dynamic Client Registration not supported" "Do you want to manually provide a client registration (client id)?"

Reply:
If possible, include that `http://localhost:*`, `https://vscode.dev/redirect`, should be registered as redirect uris.